buildSrc/src/main/java/org/springframework/boot/build/KotlinConventions.java

```diff
@@ -46,9 +46,9 @@ void apply(Project project) {
 		project.getPlugins().withId("org.jetbrains.kotlin.jvm", (plugin) -> {
 			project.getTasks().withType(KotlinCompile.class, (compile) -> {
 				KotlinJvmOptions kotlinOptions = compile.getKotlinOptions();
-				kotlinOptions.setApiVersion("1.3");
-				kotlinOptions.setLanguageVersion("1.3");
-				kotlinOptions.setAllWarningsAsErrors(true);
+				kotlinOptions.setApiVersion("1.6");
+				kotlinOptions.setLanguageVersion("1.6");
+				kotlinOptions.setAllWarningsAsErrors(false);
```

Comment:
I took the liberty to switch this to `false` because there are warnings that are meaningless in sample code (such as `parameter is never used` in functions with empty bodies) and they break the build.

Reply:
We should do this specifically for the docs project, leaving warnings as errors by convention.

Reply:
Returned this setting to `true`; configured `false` in `build.gradle` of spring-boot-docs.